src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -284,16 +460,30 @@ public void serializeToView(@NotNull final ByteBuf buf)
         super.serializeToView(buf);
 
         buf.writeInt(worker == null ? 0 : worker.getId());
+
+        buf.writeInt(recipes.size());
+        for(final IToken token: recipes)
+        {
+            ColonyManager.getRecipes().get(token).writeToBuffer(buf);
+        }
     }
 
     /**
      * AbstractBuildingWorker View for clients.
      */
     public static class View extends AbstractBuildingHut.View
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 59 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)